services/diddoc/echo_handlers.go

```diff
@@ -118,3 +141,29 @@ func DidDocVersionMetadataEchoHandler(c echo.Context) error {
 func DidDocAllVersionMetadataEchoHandler(c echo.Context) error {
 	return services.EchoWrapHandler(&DIDDocAllVersionMetadataRequestService{})(c)
 }
+
+// DidDocResourceCollectionEchoHandler godoc
+//
+//	@Summary		Fetch metadata for all Resources
+//	@Description	Get metadata for all Resources within a DID Resource Collection
+//	@Tags			Resource Resolution
+//	@Accept			application/did+ld+json,application/ld+json,application/did+json
+//	@Produce		application/did+ld+json,application/ld+json,application/did+json
+//	@Param			did	path		string	true	"Full DID with unique identifier"
+//	@Success		200	{object}	types.ResourceDereferencing{contentStream=types.ResolutionDidDocMetadata}
+//	@Failure		400	{object}	types.IdentityError
+//	@Failure		404	{object}	types.IdentityError
+//	@Failure		406	{object}	types.IdentityError
+//	@Failure		500	{object}	types.IdentityError
+//	@Failure		501	{object}	types.IdentityError
+//	@Router			/{did}/metadata [get]
+func DidDocResourceCollectionEchoHandler(c echo.Context) error {
+	// Get Accept header
+	acceptHeader := c.Request().Header.Get(echo.HeaderAccept)
+	requestedContentType, profile := services.GetPriorityContentType(acceptHeader, false)
+	didParam := c.Param("did")
+	if !requestedContentType.IsSupported() {
+		return types.NewRepresentationNotSupportedError(didParam, requestedContentType, nil, false)
+	}
+	return services.EchoWrapHandler(&DIDDocResourceDereferencingService{Profile: profile})(c)
+}
```

Comment:
When would this be triggered?

Reply:
This handles `<did>/metadata` endpoint

Reply:
Name it more sensibly if you're moving it